src/app_service/service/message/service.nim

```diff
@@ -222,6 +227,37 @@ QtObject:
       reactionId: r.id, reactionFrom: r.`from`)
       self.events.emit(SIGNAL_MESSAGE_REACTION_FROM_OTHERS, data)
 
+  proc handleMessagesReload(self: Service, communityId: string) =
+    var keys = newSeq[string]()
+    for k in self.msgCursor.keys:
+      if k.startsWith(communityId):
+        keys.add(k)
+    for k in keys:
+      self.msgCursor.del(k)
+
+    keys = @[]
+    for k in self.lastUsedMsgCursor.keys:
+      if k.startsWith(communityId):
+        keys.add(k)
+    for k in keys:
+      self.lastUsedMsgCursor.del(k)
+
+    keys = @[]
+    for k in self.pinnedMsgCursor.keys:
+      if k.startsWith(communityId):
+        keys.add(k)
+    for k in keys:
+      self.pinnedMsgCursor.del(k)
+
+    keys = @[]
+    for k in self.lastUsedPinnedMsgCursor.keys:
+      if k.startsWith(communityId):
+        keys.add(k)
+    for k in keys:
+      self.lastUsedPinnedMsgCursor.del(k)
```

Comment:
Something tells me there is a way to make this cooler with a macro or something, but it's not that worth it haha.